phys2bids/physio_obj.py

```diff
@@ -162,17 +251,103 @@ def ch_amount(self):
+        idx: int, tuple of int or slicer
+            indexes to use to slice the timeseries.
+
+        Returns
+        -------
+        BlueprintInput object
+            a copy of the object with the part of timeseries expressed by idx.
+
+        Raises
+        ------
+        IndexError
+            If the idx, represented as a slice, is out of bounds.
+
+        Notes
+        -----
+        If idx is an integer, it returns an instantaneous moment for all channels.
+        If it's a slicing, it always return the full slice. This means that
+        potentially, depending on the frequencies, BlueprintInput[1] and
+        BlueprintInput[1:2] might return different results.
+        """
+        sliced_timeseries = [None] * self.ch_amount
+        return_instant = False
+        trigger_length = len(self.timeseries[self.trigger_idx])
+
+        if not self.trigger_idx:
+            self.trigger_idx = 0
+
+        # If idx is an integer, return an "instantaneous slice" and initialise slice
+        if isinstance(idx, int):
+            return_instant = True
+            if idx < 0:
+                idx = trigger_length + idx
+
+            idx = slice(idx, idx + 1)
+
+        if idx.start >= trigger_length or idx.stop > trigger_length:
+            raise IndexError(f'slice ({idx.start}, {idx.stop}) is out of '
+                             f'bounds for channel {self.trigger_idx} '
+                             f'with size {trigger_length}')
+
+        # Operate on each channel on its own
+        for n, channel in enumerate(self.timeseries):
+            idx_dict = {'start': idx.start, 'stop': idx.stop, 'step': idx.step}
+            # Adapt the slicing indexes to the right requency
+            for i in ['start', 'stop', 'step']:
+                if idx_dict[i]:
+                    idx_dict[i] = int(np.floor(self.freq[n]
+                                               / self.freq[self.trigger_idx]
+                                               * idx_dict[i]))
+
+            # Correct the slicing stop if necessary
+            if idx_dict['start'] == idx_dict['stop'] or return_instant:
+                idx_dict['stop'] = idx_dict['start'] + 1
+            elif trigger_length == idx.stop:
+                idx_dict['stop'] = len(channel)
+
+            new_idx = slice(idx_dict['start'], idx_dict['stop'], idx_dict['step'])
+            sliced_timeseries[n] = channel[new_idx]
+
+        return BlueprintInput(sliced_timeseries, self.freq, self.ch_name,
+                              self.units, self.trigger_idx)
+
+    def __eq__(self, other):
+        """
+        Return test of equality between two objects.
+
+        Parameters
+        ----------
+        other:
+            comparable object.
+
+        Returns
+        -------
+        boolean
         """
-        Renames the channels. If 'time' or 'trigger' were specified,
-        it makes sure that they're the first and second entry.
+        return are_equal(self, other)
```

Comment:
Is this function really needed? It only calls the `are_equal( )` function. It does nothing else.

Reply:
Okay, I see now this is part of the class and the `are_equal( )` function isn't. I'm sorry for that. You know how GitHub hides stuff in reviews.